src/library/export/trackexportdlg.cpp

```diff
@@ -48,6 +52,45 @@ void TrackExportDlg::showEvent(QShowEvent* event) {
         qDebug() << "Programming error: did not initialize m_exporter, about to crash";
         return;
     }
+
+    //Checks if there are any files that are missing. If so, it will throw dialog informing the user.
+    //User can then decide to either cancel the export or skip those files
+    const QList<QString> files = m_worker->getMissingTracks();
+    if (!files.isEmpty()) {
+        QDialog notExportedDlg;
+        notExportedDlg.setWindowTitle(tr("Export Track Files"));
+        QVBoxLayout* notExportedLayout = new QVBoxLayout;
+        QLabel* notExportedLabel = new QLabel;
+        notExportedLabel->setText(
+                tr("The following %1 files were not found at the specified "
+                   "file location and as a result were not exported."
+                   "Click \"OK\" to skip these files. Click \"Cancel\" to cancel the export.")
+                        .arg(QString::number(files.length())));
+        notExportedLabel->setTextFormat(Qt::RichText);
+
+        QListWidget* notExportedList = new QListWidget;
+        notExportedList->addItems(files);
+
+        QDialogButtonBox* exportedDlgButtons = new QDialogButtonBox();
+        exportedDlgButtons->addButton(QDialogButtonBox::Ok);
+        exportedDlgButtons->addButton(QDialogButtonBox::Cancel);
+        connect(exportedDlgButtons,
+                &QDialogButtonBox::accepted,
+                &notExportedDlg,
+                &QDialog::accept);
+        connect(exportedDlgButtons,
+                &QDialogButtonBox::rejected,
+                &notExportedDlg,
+                &QDialog::reject);
+
+        notExportedLayout->addWidget(notExportedLabel);
+        notExportedLayout->addWidget(notExportedList);
+        notExportedLayout->addWidget(exportedDlgButtons);
+        notExportedDlg.setLayout(notExportedLayout);
+        if (notExportedDlg.exec() == QDialog::Rejected) {
+            m_worker->m_canceledByUser = true;
```

Comment:
Why do we even start the worker her? 
How about just 
hide();
accept();
without bothering the user with additional pop ups?

Reply:
i tried to use hide and accept, instead of having the current setup, but it didn't seem to work. would always end up with a blank box that couldn't be closed